### PR TITLE
Use "ingestion" instead of "sender" throughout the SDK

### PR DESCRIPTION
**Pull request by TroubleMakerBen**
- Rename `*sender*` classes, protocols and enums
- Rename methods containing `*sender*`
- Fix comments containing `*sender*`

### CODE REVIEW

File: AppCenterPush/AppCenterPush/MSPush.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * Defines the class that implements the optional protocol `MSPushDelegate`.
  *
- * @param delegate Sender's delegate.
+ * @param delegate The delegate.
```
Comment (clpolet) [inline AppCenterPush/AppCenterPush/MSPush.h]: Good catch ;)

File: AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m [hunk 2]
```diff
@@ -72,20 +72,20 @@ - (void)removeDelegate:(id<MSChannelDelegate>)delegate {
   });
 }
 
-#pragma mark - MSSenderDelegate
+#pragma mark - MSIngestionDelegate
 
-- (void)senderDidSuspend:(id<MSSender>)sender {
-  (void)sender;
+- (void)ingestionDidSuspend:(id<MSIngestionProtocol>)ingestion {
```
[the remark this answers is not in the document]
Comment (TroubleMakerBen) [reply AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m]: I'll create a separate PR for that.

[the remark this answers is not in the document]
Comment (jaelim-ms) [reply AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m]: #1042 contains all of those changes + this change. Are you going to get both PRs in?

File: AppCenter/AppCenterTests/MSAppCenterIngestionTests.m [hunk 3]
```diff
@@ -22,10 +22,11 @@ @interface MSAppCenterIngestionTests : XCTestCase
 
 @end
 
-// TODO: Separate base MSHttpSender tests from this test and instantiate MSAppCenterIngestion with initWithBaseUrl:, not
-// the one with multiple parameters.
-// Look at comments in each method.
-// Add testHeaders to verify headers are populated properly. Look at testHeaders in MSOneCollectorIngestionTests
+/*
+* TODO: Separate base MSHttpIngestion tests from this test and instantiate MSAppCenterIngestion with initWithBaseUrl:,
```
Comment (jaelim-ms) [inline AppCenter/AppCenterTests/MSAppCenterIngestionTests.m]: Reformat to align * in the comment

File: AppCenter/AppCenter/Internals/Ingestion/MSIngestionDelegate.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@protocol MSIngestionDelegate <NSObject>
+
+@optional
+
+/**
+ * Triggered after the ingestion has suspended its state.
+ *
+ * @param ingestion Ingestion.
+ */
+- (void)ingestionDidSuspend:(__unused id<MSIngestionProtocol>)ingestion;
```
Comment (clpolet) [inline AppCenter/AppCenter/Internals/Ingestion/MSIngestionDelegate.h]: I think we should not imply that the parameter won't be used by the delegates here.

**Pull request merged**